src/dkg/key_generation.rs

```diff
@@ -911,23 +933,27 @@ impl<C: CipherSuite> DistributedKeyGeneration<RoundOne, C> {
                     }
                 }
                 if let Ok(share) = decrypted_share {
-                    my_secret_shares.push(share);
+                    my_secret_shares.insert(share.sender_index, share);
                 }
             } else {
                 return Err(Error::Custom("to_round_two() was called with encrypted secret shares containing invalid indices".to_string()));
             }
         }
 
-        if !complaints.is_empty() {
+        // If there are less than `t` participants remaining (i.e. honest) we can only abort.
+        if self.state.parameters.n - (complaints.len() as u32) < self.state.parameters.t {
```

Comment:
This is something that puzzled me for a bit, whether we should update `n` on the go to reflect the current list of available participants, or if it should be done outside of this library. I would tend to prefer the former, as it would remove room for error, but I'd understand the will to keep the original parameters reflected. Not sure what's best, any opinion?

Reply:
Hmm, I have a slight preference for updating `n` and return it at the end too, but I don't have a strong opinion either. We can ask the others.

Reply:
@LindaGuiga @4l0n50 any opinion on this? Should we update parameters `n` along the DKG phases to always list the remaining participants number? Or should we keep the original `n` and maintain a tracking `remaining_participants_count`?

Reply:
Addressed!